oaknational/Oak-Web-Application
# fix: hubspot load script

## Description

- Use `next/script` to load hubspot tracking script
- Create `<ConditionalScript shouldLoad={} ... />`  which wraps next/script component 
- Use onError to handle any load errors

## Issue(s)

Fixes #999

## How to test

1. Go to {cloud link}, analytics consent enabled
2. with dev tools open, should see script loaded (and subsequent hubspot scripts)
3.  should **not** see any console errors relating to leadin flows or hubspot in general
4. **NB** the console errors are not reproducible (only occur some of the time), so likely this is a case of having to merge it and wait to check if BugSnag clears up before considering this "done"

=======================

1. WIth analytics consent disabled
2. with dev tools open, shoud **not** see any hubspot scripts load
